docs/concepts/typings.md

```diff
@@ -4,4 +4,103 @@ title: Type hints
+
+A complete example of this, extending upon [Schema basics](https://strawberry.rocks/docs/general/schema-basics), might be the following:
+
+```python
+import datetime
+import typing
+import strawberry
+
+BOOKS_LOOKUP = {
+    'Frank Herbert': [{
+        'title': 'Dune',
+        'date_published': '1965-08-01',
+        'price': '5.99',
+        'isbn': 9780801950773
+    }],
+}
+
+@strawberry.type
+class Book:
+    title: str
+    author: 'Author'
+    date_published: datetime.date
+    price: decimal.Decimal
+    isbn: Union[int, str] # could be 1234567890000 or '123-4-56-789000-0'
+
+def get_books_by_author(root: 'Author') -> typing.List['Book']:
+    # error will be thrown in here if a book in BOOKS_LOOKUP is missing a field
+    # since none of the types of Book are optional
+    stored_books: typing.List[typing.Dict[str, str | int]] = BOOKS_LOOKUP[root.name]
+    return [Book(
+        title = book.get('title'),
+        author = root,
+        date_published = book.get('date_published'),
+        price = book.get('price'),
+        isbn = book.get('isbn')
+    ) for book in stored_books]
+
+@strawberry.type
+class Author:
+    name: str
+    books: typing.List['Book'] = strawberry.field(resolver=get_books_by_author)
+
+@strawberry.type
+class Group:
+    name: Optional[str] # groups of authors don't necessarily have names
+    authors: typing.List['Author']
+
+    @strawberry.field
+    def books(self) -> typing.List['Book']:
+        books = []
+        for author in self.authors:
+            books += get_books_by_author(author)
+        return books
+
+```
+
+- `self` within a resolver's definition, whether decorated as `@strawberry.field` or `@strawberry.mutation`, never needs a type hint because it can be inferred.
+- `Optional` is the way to tell Strawberry that a field is nullable. Without it, every field is assumed to be non-null. This is in contrast to graphene wherein every field is assumed nullable unless `required=True` is supplied.
+- Type hinting doesn't stop at being a requirement for Strawberry to function, it is also immensely helpful for collaborating developers. By specifying the type of `stored_books` in `get_books_by_author`, an IDE equipped with PyLance will be able to infer that `book` within the list comprehension is a dictionary and so will understand that `.get()` is a method function of the `dict` class. This helps the readability and maintainability of written code.
+
+## Motivation
+
+Python, much like Javascript and Ruby, is a _dynamically typed_ language that allows for high-level programming where the fundamental types of variables, e.g. integers, arrays, hash-maps, _etc._, are understood by the machine at _runtime_ through Just-in-Time compilation.
+
+Yet, much like the low-level languages of C, Java, and Rust, the graphQL query language is _statically typed_ since the data types defined by the schema must be known prior to compiling the API code in order to define a definite schema to query against.
```

Comment:
```suggestion
Yet, much like the low-level languages of C, Java, and Rust, the GraphQL query language is _statically typed_ since the data types defined by the schema must be known prior to compiling the API code in order to define a definite schema to query against.
```

This needs adjustment in a couple places in this file